Add help text to create.py

Mrt134 opened the pull request:
Added help text to the -r, -t, -b, and -e options.
Rework the -f option to match the other options' syntax.
TODO - need clarification on the -c and -p options before adding that help text.

Make sure to follow and check these boxes before submitting a PR! Thank you.

- [x] Explain the feature
- [x] Read [CONTRIBUTING.md](https://github.com/iocage/iocage/blob/master/CONTRIBUTING.md)

Mrt134 committed:
Add help text to create.py

Added help text to the -r, -t, -b, and -e options.
Rework the -f option to match the other options' syntax.
TODO - need clarification on the -c and -p options before adding that help text.

skarekrow commented on iocage/cli/create.py:
```diff
@@ -53,21 +53,28 @@ def validate_count(ctx, param, value):
-@click.option("--release", "-r", required=False)
-@click.option("--template", "-t", required=False)
+@click.option("--release", "-r", required=False,
+              help="Specify the release to use for the new jail.")
```
`release` should be all caps `RELEASE` as it's specifically meant to reference the available `RELEASEs`.

Mrt134 replied:
Complete - see new commit on this PR.

skarekrow commented on iocage/cli/create.py:
```diff
@@ -53,21 +53,28 @@ def validate_count(ctx, param, value):
 
 @click.command(name="create", help="Create a jail.")
 @click.option("--count", "-c", callback=validate_count, default="1")
```
`-c` will create that many jails. So if you issue `iocage create -c 20 -r 11.0-RELEASE tag="foo"` you will get 20 jails, labeled `foo_1` to `foo_20`. I'll leave the wording up to you :P

Mrt134 replied:
Complete - see new commit on this PR.

skarekrow commented on iocage/cli/create.py:
```diff
@@ -53,21 +53,28 @@ def validate_count(ctx, param, value):
+@click.option("--template", "-t", required=False,
+              help="Flag this jail as a template, which allows for rapid "
+                   "redeployment of a customized jail.")
 @click.option("--pkglist", "-p", default=None)
```
`-p` allows you to specify a JSON file who's purpose is to install each package inside the jail after creation.

Example:
```
{
    "pkgs": [
    "foo",
    "bar"
    ]
}
``` I have an error prompt for the user if it's in an incorrect format, so perhaps just explaining the concept is enough without also showing the format of the JSON.

Mrt134 replied:
Complete - see new commit on this PR.

skarekrow commented on iocage/cli/create.py:
```diff
@@ -53,21 +53,28 @@ def validate_count(ctx, param, value):
+              help="Specify the release to use for the new jail.")
+@click.option("--template", "-t", required=False,
+              help="Flag this jail as a template, which allows for rapid "
+                   "redeployment of a customized jail.")
```
I prefer to have my spaces on the next line, so this would be:
```
              help="Flag this jail as a template, which allows for rapid"
                   " redeployment of a customized jail.")
```

Mrt134 replied:
Complete - see new commit on this PR.

skarekrow commented on iocage/cli/create.py:
```diff
@@ -53,21 +53,28 @@ def validate_count(ctx, param, value):
+              help="Provide a specific UUID for this jail.")
+@click.option("--basejail", "-b", is_flag=True, default=False,
+              help="Set the new jail type to a basejail. Basejails "
+                   "re-clone their base filesystems at each startup.")
```
This is only true for the legacy implementation. The new one actually mounts the specified `RELEASEs` directories as `nullfs` mounts over the jails directories.

So `11.0-RELEASE/root/bin` will be mounted as a R/O nullfs mount over `foo/root/bin`, saving space and making moving to a new `RELEASE` much faster. Once the upgrade mechanism is figured out and implemented that is ;)

Mrt134 replied:
Complete - see new commit on this PR.

skarekrow commented on iocage/cli/create.py:
```diff
@@ -53,21 +53,28 @@ def validate_count(ctx, param, value):
+                   "re-clone their base filesystems at each startup.")
+@click.option("--empty", "-e", is_flag=True, default=False,
+              help="Create an empty jail used for unsupported or custom "
+                   "jails.")
```
Same space comment as above.

Mrt134 replied:
Complete - see new commit on this PR.

Mrt134 committed:
Add in missing help text and skarekrow's fixes:

Add description of -c, --count.
Capitalize RELEASE instances.
Adjust word spacing when help text flows over a single line.
Add description of -p, --pkglist.
Rework the -b, --basejail help text.

skarekrow commented:
Thanks! :+1:
